Dockerfile.openshift

```diff
@@ -0,0 +1,12 @@
+FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.16-openshift-4.9 AS builder
+WORKDIR /external-dns-operator
+COPY . .
+RUN make buildbin
```

Comment:
Should we make this change for `Dockerfile` as well? I don't have a strong preference between using the Makefile's compile call, or an explicit call to `go build` in the dockerfiles, but I do have a preference for consistency.